src/functions/sendCandidateResults/application/service/categories/BE/__tests__/fault-provider-cat-be.spec.ts

```diff
@@ -0,0 +1,218 @@
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
+import { CompetencyOutcome } from '../../../../../domain/competency-outcome';
+import { Fault } from '../../../../../domain/fault';
+import {
+  getDangerousFaultsCatBE,
+  getDrivingFaultsCatBE,
+  getNonStandardFaultsCatBE,
+  getSeriousFaultsCatBE,
+  getVehicleChecksFaultCatBE,
+} from '../fault-provider-cat-be';
+import { Competencies } from '../../../../../domain/competencies';
+
+describe('fault-provider-cat-be', () => {
+  describe('getVehicleChecksFaultCatBE', () => {
+    it('should find a dangerous fault if one exists', () => {
+      const data: CatBEUniqueTypes.VehicleChecks = {
+        showMeQuestions: [{
+          outcome: CompetencyOutcome.D,
+        }],
+      };
+      const result: Fault[] = getVehicleChecksFaultCatBE(data, CompetencyOutcome.D);
+      expect(result.length).toBe(1);
+      expect(result).toContain({ name: Competencies.vehicleChecks, count: 1 });
+    });
+
+    it('should not find a dangerous fault if one exists', () => {
+      const data: CatBEUniqueTypes.VehicleChecks = {
+        showMeQuestions: [{
+          outcome: CompetencyOutcome.DF,
+        }],
+      };
+      const result: Fault[] = getVehicleChecksFaultCatBE(data, CompetencyOutcome.D);
+      expect(result.length).toBe(0);
+    });
+
+    it('should find a serious fault if one exists', () => {
+      const data: CatBEUniqueTypes.VehicleChecks = {
+        showMeQuestions: [{
+          outcome: CompetencyOutcome.S,
+        }],
+      };
+      const result: Fault[] = getVehicleChecksFaultCatBE(data, CompetencyOutcome.S);
+      expect(result.length).toBe(1);
+      expect(result).toContain({ name: Competencies.vehicleChecks, count: 1 });
+    });
+
+    it('should not find a serious fault if one exists', () => {
+      const data: CatBEUniqueTypes.VehicleChecks = {
+        showMeQuestions: [{
+          outcome: CompetencyOutcome.DF,
+        }],
+      };
+      const result: Fault[] = getVehicleChecksFaultCatBE(data, CompetencyOutcome.S);
+      expect(result.length).toBe(0);
+    });
+
+    it('should find a driving fault if one exists', () => {
+      const data: CatBEUniqueTypes.VehicleChecks = {
+        showMeQuestions: [{
+          outcome: CompetencyOutcome.DF,
+        }],
+      };
+      const result: Fault[] = getVehicleChecksFaultCatBE(data, CompetencyOutcome.DF);
+      expect(result.length).toBe(1);
+      expect(result).toContain({ name: Competencies.vehicleChecks, count: 1 });
+    });
+
+    it('should not find a serious fault if one exists', () => {
+      const data: CatBEUniqueTypes.VehicleChecks = {
+        showMeQuestions: [{
+          outcome: CompetencyOutcome.S,
+        }],
+      };
+      const result: Fault[] = getVehicleChecksFaultCatBE(data, CompetencyOutcome.DF);
+      expect(result.length).toBe(0);
+    });
```

Comment:
Not sure this logic applies. Also test looks dodgy - you create a serious fault but look for a driving fault, so it won't find anything?